### ISSUE TITLE
[Bug][config-ui] Gitlab parentId can conflict with project Ids (recursion issue in ui)

### ISSUE THREAD

**Issue by amanpatel** [apache/incubator-devlake#4992]
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-devlake/issues?q=is%3Aissue) and found no similar issues.

### What happened

I am facing an issue of "maximum call stack size exceeded" in a very special/narrow case. This happens on a brand new project, and while adding a blueprint/data connection.

The details of the reproduction and the cause of the issue are all explained in the linked video on slack.

### What do you expect to happen

No recursion error.

### How to reproduce

See the video linked here: https://devlake-io.slack.com/files/U04SMMED8AY/F0514M4R5DH/video2178565862.mp4

### Version

v0.16.0

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

**Comment by klesh** [comment]
@amanpatel Thanks for your report, very thorough! will look into it ASAP

**Comment by mintsweet** [comment]
hi @amanpatel, could you please provide me with your data, I want to reproduce it locally.

**Comment by amanpatel** [comment]
@mintsweet  Yes

Request URL: http://(domain)/api/plugins/gitlab/connections/3/remote-scopes?groupId=5

Response JSON (I have modified it to remove sensitive details). The first entry in the "children" array is the one that causes the problem, as it has id:5 and parentId of 5 (recursive).

In gitlab the parentID is not the project but a group id. So they are allowed to collide...

```
{
  "children": [

    {
      "type": "scope",
      "parentId": "5",
      "id": "5",
      "name": "Networks API",
      "data": {
        "connectionId": 0,
        "gitlabId": 5,
        "name": "Networks API",
        "description": ".",
        "defaultBranch": "develop",
        "pathWithNamespace": "resonance/networks-api",
        "webUrl": "https://.../resonance/networks-api",
        "creatorId": 0,
        "visibility": "private",
        "openIssuesCount": 0,
        "starCount": 2,
        "forkedFromProjectId": 0,
        "forkedFromProjectWebUrl": "",
        "httpUrlToRepo": "https://.../resonance/networks-api.git",
        "createdDate": "2021-06-15T22:12:31.916Z",
        "updatedDate": "2023-04-22T00:25:24.737Z"
      }
    },
    {
      "type": "scope",
      "parentId": "5",
      "id": "4",
      "name": "Networks Frontend",
      "data": {
        "connectionId": 0,
        "gitlabId": 4,
        "name": "Networks Frontend",
        "description": ".",
        "defaultBranch": "develop",
        "pathWithNamespace": "resonance/networks-frontend",
        "webUrl": "https://.../resonance/networks-frontend",
        "creatorId": 0,
        "visibility": "private",
        "openIssuesCount": 0,
        "starCount": 2,
        "forkedFromProjectId": 0,
        "forkedFromProjectWebUrl": "",
        "httpUrlToRepo": "https://.../resonance/networks-frontend.git",
        "createdDate": "2021-06-15T22:04:43.105Z",
        "updatedDate": "2023-04-25T01:30:04.915Z"
      }
    }
  ],
  "nextPageToken": ""
}
```

**Comment by klesh** [comment]
@amanpatel Thanks for the information, it should be fixed by https://github.com/apache/incubator-devlake/pull/5037
and will be available on `v0.17` in 2 weeks.
